compiler/ballerina-parser/src/main/java/io/ballerinalang/compiler/internal/parser/DocumentationLexer.java

```diff
@@ -669,23 +702,79 @@ private STToken readDocumentationBacktickContentToken() {
         if (nextToken == LexerTerminals.BACKTICK) {
             reader.advance();
             switchMode(ParserMode.DOCUMENTATION_INTERNAL);
+            resetSpecialKeywordInPlace();
             return getDocumentationSyntaxTokenWithNoTrivia(SyntaxKind.BACKTICK_TOKEN);
         }
 
+        int lookAheadCount = 0;
+        String backtickStr = "";
         while (!reader.isEOF()) {
             switch (nextToken) {
                 case LexerTerminals.BACKTICK:
                 case LexerTerminals.NEWLINE:
                 case LexerTerminals.CARRIAGE_RETURN:
                     break;
                 default:
-                    reader.advance();
-                    nextToken = peek();
+                    backtickStr = backtickStr.concat(String.valueOf((char) nextToken));
+                    lookAheadCount++;
+                    nextToken = reader.peek(lookAheadCount);
                     continue;
             }
             break;
         }
 
-        return getDocumentationLiteral(SyntaxKind.BACKTICK_CONTENT);
+        if (!isSpecialKeywordInPlace) {
+            // Look for a x(), m:x(), T.y(), m:T.y() match
+            boolean hasMatch = backtickStr.matches(backtickExprRegex);
+            if (hasMatch) {
+                switchMode(ParserMode.DOCUMENTATION_BACKTICK_EXPR);
+                return readDocumentationBacktickExprToken();
+            } else {
+                reportLexerError(DiagnosticWarningCode.WARNING_INVALID_DOCUMENTATION_EXPRESSION);
+                reader.advance(lookAheadCount);
+                return getDocumentationLiteral(SyntaxKind.BACKTICK_CONTENT);
+            }
+        }
+
+        reader.advance(lookAheadCount);
+        resetSpecialKeywordInPlace();
+
+        boolean isIdentifier = backtickStr.matches(identifierRegex);
+        if (isIdentifier) {
+            return getIdentifierToken();
+        } else {
+            reportLexerError(DiagnosticWarningCode.WARNING_INVALID_DOCUMENTATION_IDENTIFIER);
```

Comment:
do we inject parser diagnostic to the compiler diagnostic somewhere? or two APIs act independently?
right now I log a warning for this in the node transformer as well.

Reply:
Yes, we add all the parser errors to the compiler-diagnostics at the [Parser.java](https://github.com/ballerina-platform/ballerina-lang/blob/74c3396849e62f2b13b587bab6af3d1cd8cf7be9/compiler/ballerina-lang/src/main/java/org/wso2/ballerinalang/compiler/parser/Parser.java#L265) class